### PR TITLE
Set of fixes for cpuload calculations.

### PR DESCRIPTION
**Pull request by rjnagal**
. Remove counting of taskgroups from scheddebug.
. Move monitoring thread 500ms ahead of other containers housekeeping.
. Rely on /proc/loadavg for root load.
. Cover up for scheddebug atomicity issues (WIP)
. Remove counting of monitoring thread.

Getting better, but still a bit farther away from ideal load :(

### CODE REVIEW

File: utils/cpuload/scheddebug/scheddebug.go [hunk 1]
```diff
@@ -66,6 +69,7 @@ func (self *SchedReader) Stop() {
 // there is a chance of sometimes picking the last cycle's data. We can solve that by
 // calling this housekeeping from globalhousekeeping if its an issue.
 func (self *SchedReader) housekeep() {
+	time.Sleep(500 * time.Millisecond)
```
Comment (vmarmol) [inline utils/cpuload/scheddebug/scheddebug.go]: nit: Add a doc as to why we're doing this

File: utils/cpuload/scheddebug/scheddebug.go [hunk 2]
```diff
@@ -142,9 +178,52 @@ func (self *SchedReader) allowErrorLogging() bool {
 	return false
 }
 
+func GetSelfCgroup() (string, error) {
```
Comment (vmarmol) [inline utils/cpuload/scheddebug/scheddebug.go]: We already get this information at the start of cAdvisor. Worth it to re-use it here?

File: utils/cpuload/scheddebug/scheddebug.go [hunk 3]
```diff
@@ -142,9 +178,52 @@ func (self *SchedReader) allowErrorLogging() bool {
...
+	numRunning, err := strconv.ParseInt(string(matches[1]), 10, 64)
+	if err != nil {
+		return -1, fmt.Errorf("could not parse number of running processes from %q: %v", matches[1], err)
+	}
+	numRunning -= 1
+	return numRunning, nil
+}
+
+// Return parent cgroup name given an absolute cgroup name.
+func GetParent(c string) string {
```
Comment (vmarmol) [inline utils/cpuload/scheddebug/scheddebug.go]: nit: not public?

File: utils/cpuload/scheddebug/scheddebug.go [hunk 4]
```diff
@@ -142,9 +178,52 @@ func (self *SchedReader) allowErrorLogging() bool {
...
+		return "", fmt.Errorf("failed to read cgroup path for cAdvisor: %v", err)
+	}
+	matches := selfCgroupRegExp.FindSubmatch(out)
+	if len(matches) != 2 {
+		return "", fmt.Errorf("could not find cpu cgroup path in %q", string(out))
+	}
+	return string(matches[1]), nil
+}
+
+func GetRootLoad() (int64, error) {
```
Comment (vmarmol) [inline utils/cpuload/scheddebug/scheddebug.go]: not public?

File: utils/cpuload/scheddebug/scheddebug.go (hunk 2 above)
Comment (vmarmol) [inline utils/cpuload/scheddebug/scheddebug.go]: not public? :)

File: utils/cpuload/scheddebug/scheddebug.go (hunk 1 above)
Comment (rjnagal) [inline utils/cpuload/scheddebug/scheddebug.go]: Added.

File: utils/cpuload/scheddebug/scheddebug.go (hunk 2 above)
Comment (rjnagal) [inline utils/cpuload/scheddebug/scheddebug.go]: We call it just once and cache it here. Wanted to keep cAdvisor dependencies on utils at a minimum.

File: utils/cpuload/scheddebug/scheddebug.go (hunk 3 above)
Comment (rjnagal) [inline utils/cpuload/scheddebug/scheddebug.go]: Gah updates all the caps!

**Comment by vmarmol** [comment]
LGTM

**Pull request merged**